tests/test_pagination.py

```diff
@@ -152,16 +149,66 @@ def pagination_test(self):
+                                                   if actual_record.get(pk) == record.get(pk)]
+                        self.assertTrue(len(stream_expected_records),
+                                        msg="An actual record is missing from our expectations: \nRECORD: {}".format(actual_record))
+                        self.assertEqual(1, len(stream_expected_records),
+                                         msg="A duplicate record was found in our expectations for {}.".format(stream))
+                        stream_expected_record = stream_expected_records[0]
+                        self.assertEqual(stream_expected_record, actual_record.get(pk))
+
+                    # Verify that our expected records were replicated by the tap
+                    for expected_record in expected_records.get(stream):
+                        stream_actual_records = [record.get(pk) for record in actual_records
+                                                 if expected_record.get(pk) == record.get(pk)]
+                        self.assertTrue(len(stream_actual_records),
+                                        msg="An expected record is missing from the sync: \nRECORD: {}".format(expected_record))
+                        self.assertEqual(1, len(stream_actual_records),
+                                         msg="A duplicate record was found in the sync for {}.".format(stream))
+                        stream_actual_record = stream_actual_records[0]
+                        self.assertEqual(expected_record.get(pk), stream_actual_record)
+
+                elif stream == 'inventories': # We can not compare by pks for this stream so we make a less strict assertion
+
+                    # Verify the expected number of records were replicated
+                    self.assertEqual(len(expected_records.get(stream)), len(actual_records))
+
+                    # Verify the pages contain unique data sets (verify we don't duplicate pages)
+                    page_size = self.API_LIMIT.get(stream)
+                    num_full_pages = len(actual_records) // page_size
+                    num_pages = num_full_pages if len(actual_records) % page_size == 0 else num_full_pages + 1
+
+                    pages_to_compare = []
+                    for page in range(num_pages):  # break the data into pages
+                        start = page*page_size
+                        end = start + page_size - 1
+                        pages_to_compare.append(actual_records[start:end])
+
+                    instances = 0
+                    for page_data in pages_to_compare: # count instances of a page of data
+                        for page in pages_to_compare:
+                            if page_data == page:
+                                instances += 1
+
+                        # Verify there is only 1 instance of a given page of data in the target
+                        self.assertEqual(1, instances, msg="There are duplicate pages of data")
```

Comment:
I do not understand what this is asserting. Why are we doing all this work to break things into pages?

Reply:
The test case is `Verify the pages contain unique data sets (verify we don't duplicate pages)`. Dan or someone had mentioned at some point that there was a tap that was paginating but was returning the same page multiple times.